code/game/machinery/computer/security.dm

```diff
@@ -521,24 +179,106 @@ What a mess.*/
 		if(prob(10/severity))
 			switch(rand(1,6))
 				if(1)
-					R.fields["name"] = "[pick(pick(GLOB.first_names_male), pick(GLOB.first_names_female))] [pick(GLOB.last_names)]"
+					R.fields[MOB_NAME] = "[pick(pick(GLOB.first_names_male), pick(GLOB.first_names_female))] [pick(GLOB.last_names)]"
 				if(2)
-					R.fields["sex"] = pick("Male", "Female")
+					R.fields[MOB_SEX] = pick(MOB_STAT_SEX_MALE, MOB_STAT_SEX_FEMALE)
 				if(3)
-					R.fields["age"] = rand(5, 85)
+					R.fields[MOB_AGE] = rand(5, 85)
 				if(4)
-					R.fields["criminal"] = pick("None", "*Arrest*", "Incarcerated", "Released", "Suspect", "NJP")
+					R.fields[MOB_CRIMINAL_STATUS] = pick(MOB_STAT_CRIME_STATUS_LIST)
 				if(5)
-					R.fields["p_stat"] = pick("*Unconscious*", "Active", "Physically Unfit")
+					R.fields[MOB_HEALTH_STATUS] = pick(MOB_STAT_HEALTH_LIST)
 				if(6)
-					R.fields["m_stat"] = pick("*Insane*", "*Unstable*", "*Watch*", "Stable")
-			continue
-
-		else if(prob(1))
-			GLOB.data_core.security -= R
-			qdel(R)
+					R.fields[MOB_MENTAL_STATUS] = pick(MOB_STAT_MENTAL_STATUS_LIST)
 			continue
 
 /obj/structure/machinery/computer/secure_data/detective_computer
 	icon = 'icons/obj/structures/machinery/computer.dmi'
 	icon_state = "messyfiles"
+
+// TGUI SEC
+// ------------------------------------------------------------------------------------------------------ //
+/obj/structure/machinery/computer/secure_data/tgui_interact(mob/user, datum/tgui/ui)
+
+	ui = SStgui.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "SecRec", name)
+		ui.open()
+
+/obj/structure/machinery/computer/secure_data/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+
+	var/mob/user = ui.user
+	playsound(src, pick('sound/machines/computer_typing4.ogg', 'sound/machines/computer_typing5.ogg', 'sound/machines/computer_typing6.ogg'), 5, 1)
+	switch(action)
+		if("authenticate")
+			var/obj/item/id_card = user.get_active_hand()
+			if (istype(id_card, /obj/item/card/id))
+				if(user.drop_held_item())
+					id_card.forceMove(src)
+					user_id_card = id_card
+			if(authenticate(user, user_id_card))
+				return TRUE
+			else
+				if(!user_id_card)
+					return
+				if(ishuman(user))
+					user_id_card.forceMove(user.loc)
+					if(!user.get_active_hand())
+						user.put_in_hands(user_id_card)
+				else
+					user_id_card.forceMove(loc)
+				user_id_card = null
+		if("logout")
+			visible_message("[SPAN_BOLD("[src]")] states, \"AUTH LOGOUT: Session end confirmed.\"")
+			authenticated = FALSE
+			if(ishuman(user))
+				user_id_card.forceMove(user.loc)
+				if(!user.get_active_hand())
+					user.put_in_hands(user_id_card)
```

Comment:
same here

Reply:
the pr is huge but you get the point

Reply:
It's really not that much, I just moved some files around to help with OOP practices.